### ISSUE TITLE
LocalizedStrings does not support clearing a translation for a locale.

### ISSUE THREAD

**Issue by clouser2** [civiform/civiform#3144]
We should add support for this. Callers are currently setting the translated text to "" when attempting this, I believe. That will trick up calls to determine if a translation is present or not (https://github.com/civiform/civiform/blob/ca0b9b70b1d366fc5c04a91a7a53697296be477a/server/app/services/LocalizedStrings.java#L119-L133).

We should likely make an explicit decision around whether an empty string implies the translation is missing. If we go that route, I'd suggest not returning `Optional`s anywhere from this API and instead always returning empty string if a key is missing.

**Comment by nb1701** [comment]
@rockycodes Was this addressed with your recent PR?

**Comment by rockycodes** [comment]
@nb1701 nope, i created a work around just for the confirmation message. This would be handy to have! My fix is here: https://github.com/civiform/civiform/blob/3e6d62e5386e08c1f7b4edc14468a6a1e35e5983/server/app/services/program/ProgramServiceImpl.java#L336 

It creates a new `LocalizedString` with just `Locale.US` set. 
```LocalizedStrings.create(ImmutableMap.of(Locale.US, ""));```

**Comment by rockycodes** [comment]
Anna got around this in her recent work to detect missing translations
